src/main/java/racingcar/racing/RacingGame.java

```diff
@@ -0,0 +1,50 @@
+package racingcar.racing;
+
+import racingcar.ui.RacingConsoleUi;
+
+import java.util.Random;
+
+public class RacingGame {
+
+    private final RacingPreference preference;
+    private final RacingEntries racingEntries;
+    private final RacingUi racingUi;
+
+
+    public RacingGame(RacingPreference preference, RacingUi racingUi, Random random) {
+        if (preference == null) {
+            throw new IllegalArgumentException("RacingPreference 값을 지정해 주세요.");
+        }
+
+        this.preference = preference;
+        this.racingEntries = new RacingEntries(this.preference.carCount(), random);
+        this.racingUi = racingUi;
+    }
+
+    public RacingEntries getRacingEntries() {
+        return this.racingEntries;
+    }
+
+    public int getRacingEntryCount() {
+        return this.racingEntries.getEntryCount();
+    }
+
+    public void start() {
+        for (int i = 0; i < preference.moveCount(); i++) {
+            racingEntries.move();
+            showStatus();
+        }
+    }
+
+    private void showStatus() {
+        racingUi.showStatus(this);
+    }
+
+
+    public static void main(String[] args) {
+        RacingUi racingUi = new RacingConsoleUi();
+        RacingPreference preference = racingUi.inputPreference();
+        RacingGame racingGame = new RacingGame(preference, racingUi, new Random());
+        racingGame.start();
```

Comment:
```suggestion
        RacingUi racingUi = new RacingConsoleUi();
        RacingPreference preference = racingUi.inputPreference();
        RacingGame racingGame = new RacingGame(preference, new Random());
        Result result = racingGame.start();
        racingUi.출력(result); 
```
입력은 RacingGame 밖에서 받는데 출력은 RacingGame 내부에서 사용하는 것이 약간 어색하네요
위와 같이 수정된다면 입출력과 비즈니스 로직을 완전히 분리시킬 수 있을 것 같아요
분리된다면 RacingGame 테스트도 쉽게할 수 있어보입니다.

_어느 시점에 출력할 것인지에 대한 제약은 없다._ 조건이 있지만
팀장님은 남달라서 리뷰드렸습니다.